### PR TITLE
feat: Improve version checker logic

### PR DESCRIPTION
**Pull request by robbie-c**
## Problem

We sometimes fix an important bug in posthog-js and want to deprecate all versions before the fix. This deprecation info currently lives in https://github.com/PostHog/posthog-js/blob/main/deprecation.json, but we don't use this for the version compatibility checker.

I also noticed that we are comparing version strings with a string compare function, rather than something semver aware.

## Changes

* Use the compatibility.json from the posthog-js repo.
* Compare version strings by parsing as them as semver, rather than doing a locale compare
* Add the version compatibility checker to web analytics
* Make it easier to have rules for the warning that depend on not just how many versions behind we are, but whether the different is major/minor/patch.

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Updated the existing tests.

### CODE REVIEW

File: frontend/src/scenes/surveys/Surveys.tsx [hunk 1]
```diff
@@ -128,7 +128,7 @@ export function Surveys(): JSX.Element {
             ) : (
                 <>
                     <div className="space-y-2">
-                        <VersionCheckerBanner minVersionAccepted="1.83.0" />
```
Comment (robbie-c) [inline frontend/src/scenes/surveys/Surveys.tsx:131]: We already deprecated this version with posthog-js's compatibility.json

**Review by benjackwhite — APPROVED** [review]
sweet!

File: frontend/src/lib/utils/semver.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
+export interface SemanticVersion {
+    major: number
+    minor?: number
+    patch?: number
+    extra?: string
+}
+
+export function parseVersion(version: string): SemanticVersion {
```
Comment (daibhin) [inline frontend/src/lib/utils/semver.ts:8]: 😍

**Review by daibhin — COMMENTED** [review]
Thanks for doing this! We get reports about the version checker being incorrect / confusing sometimes and just never had the time to take it on

**Pull request merged**